app/components/timesheet-actions.js

```diff
@@ -1,22 +1,32 @@
 import Component from '@glimmer/component';
-import { task } from 'ember-concurrency';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import constants from 'frontend-timekeeper/constants';
+const { TIMESHEET_STATUSES } = constants;
 
 export default class TimesheetActionsComponent extends Component {
-  get canSubmitHolidays() {
-    return !this.args.timesheet
-      || this.args.timesheet.isDraft;
-  }
+  @tracked absenceComplete = false;
+  @tracked timesheetComplete = false;
 
-  get canSubmitTimesheet() {
-    return !this.canSubmitHolidays
-      && this.args.timesheet.isAbsenceSubmitted;
+  @action
+  onAbsenceClick(checked) {
+    if (!checked) {
+      this.timesheetComplete = false;
+      this.args.onStatusChanged(TIMESHEET_STATUSES.DRAFT);
+    } else {
+      this.args.onStatusChanged(TIMESHEET_STATUSES.ABSENCE_SUBMITTED);
+    }
+    this.absenceComplete = checked;
   }
 
-  submitHolidays = task(async () => {
-    await this.args.onSubmitHolidays();
-  });
-
-  submitTimesheet = task(async () => {
-    await this.args.onSubmitTimesheet();
-  });
+  @action
+  onTimesheetClick(checked) {
+    if (checked) {
+      this.absenceComplete = true;
+      this.args.onStatusChanged(TIMESHEET_STATUSES.SUBMITTED);
```

Comment:
It's not the intention that users can rollback the status. They must at least be a check whether the timesheet is already exported.

Reply:
Fixed in [478117c](https://github.com/redpencilio/frontend-timekeeper/pull/4/commits/478117c8159e3e2839773477b8ae9a60702e678f)